src/test/chain-simulator/utils/test.utils.ts

```diff
@@ -0,0 +1,66 @@
+import axios from 'axios';
+import { config } from '../config/env.config';
+
+export class ChainSimulatorUtils {
+  static async waitForEpoch(targetEpoch: number = 2, maxRetries: number = 50) {
+    try {
+      // First check if simulator is running
+      await this.checkSimulatorHealth(maxRetries);
+
+      let retries = 0;
+      while (retries < maxRetries) {
+        try {
+          // Generate blocks until target epoch
+          await axios.post(
```

Comment:
we should check the epoch before generating block until epoch reached.
you can do this by fetching the `erd_epoch_number` from `${config.chainSimulatorUrl}/network/status/4294967295`

Reply:
done

Reply:
done